### PR TITLE
Fix Run Command and Demo Project

### LINKED ISSUE #1225: Fix run suggestion in metadata panel to include namespace
## Description
The metadata panel will suggest a run command when a node in the pipeline is suggested. However, if the node is within a namespace, this suggestion will not work.

Important note: The formatting of the command is also inconsistent with other Kedro resources. As a result of https://github.com/kedro-org/kedro/issues/2014 command suggestions should now follow the format `kedro <command> --<flag>=value1,value2`.

## Steps to Reproduce

1. Run `kedro run` on a new spaceflights project
2. Run `kedro viz`
3. Open metadata panel
Screenshot 2023-01-19 at 16 03 10

4. Run the suggestion under "Run Command:"

## Expected Result
The pipeline should run until the specified node.

## Actual Result
The pipeline throws an error.

```
ValueError: Pipeline does not contain nodes named ['split_data_node']. Did you mean: 
['data_science.active_modelling_pipeline.split_data_node', 
'data_science.candidate_modelling_pipeline.split_data_node']?
```

## Your Environment
Kedro version: 0.18.4
Kedro Viz version: 5.2.0

### PR DESCRIPTION
**Pull request by rashidakanchwala**
Signed-off-by: Rashida Kanchwala <rashida_kanchwala@mckinsey.com>

## Description

This is a small PR that resolves #1225 and #1243 and also adds transcoded datasets to demo-project

## Development notes

- The issue related to MemoryDatasets showing up for different nodes instead of the correct datasets  was because of incorrect naming of the dataset in the catalog.yml file. This has been fixed. 
- I have also added a simple transcoded dataset example for us to test any edge cases related to transcoding on demo project 
- For the run-command issue, all that was required was to add namespace to task nodes when doing the run_command and remove the double quotes. 

Also, currently on kedro-viz, if the user does not specify any 'name' argument for  task node, the run command on metadata panel displays the below error message. 
_'Please provide a name argument for this node in order to see a run command.'_ 

I think this bit can be improved and if there is no name argument, we could possible use the _kedro generated name_ . Currently because of this PR [https://github.com/kedro-org/kedro/pull/2139] it still won't work but once this is released in Kedro 0.18.5 -- we can do the above. 

## QA notes

Open Kedro-viz , expand ingestion and click on any task node. You will see run command show, copy that run_command. Next go to demo-projects on cli and paste the run command and see it run successfully. 

i added an additional transcoded dataset '@ pandas3' just to test the FE. i will remove it before merge as it doesn't make sense for the demo project. 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: src/components/metadata/styles/metadata.scss [hunk 1]
```diff
@@ -193,18 +193,13 @@
...
-
-  li {
-    display: inline;
-    margin: $list-inline-spacing $list-inline-spacing * 1.5;
-  }
 }
 
 .pipeline-metadata__value-list--commas li:not(:last-child):after {
-  content: ', ';
+  content: ' , ';
```
Comment (tynandebold) [inline src/components/metadata/styles/metadata.scss:202]: I don't think we need that space before the comma.

```suggestion
  content: ', ';
```

File: src/components/metadata/styles/metadata.scss [hunk 2]
```diff
@@ -193,18 +193,13 @@
 $list-inline-spacing: 0.2em;
 
 .pipeline-metadata__value-list--inline {
-  margin: 0 (-$list-inline-spacing * 1.5);
   padding: 0;
+  line-height: 1.5em;
```
Comment (tynandebold) [inline src/components/metadata/styles/metadata.scss:197]: I think we need a `margin: 0;` here. The distance between `Transcoded Types:` and the line below is greater than the distance between `Original Type:` and the line below:

image

**Review by tynandebold — APPROVED** [review]
Looks great. Approving with only a couple of small comments.

The run command seems to work fine now as well. I tested it on the demo project and I didn't get the error that @AhdraMeraliQB reported in her issue.

One other thought: can you disable the copy-to-clipboard button in this scenario? 

image

I don't think a user would want to copy that text.

File: src/components/metadata/metadata.js [hunk 3]
```diff
@@ -203,7 +201,10 @@ const MetaData = ({
                   />
                 )}
                 <MetaDataRow label="Run Command:" visible={Boolean(runCommand)}>
-                  <CommandCopier command={runCommand} />
+                  <CommandCopier
+                    command={runCommand}
+                    isCommand={metadata?.runCommand}
```
Comment (tynandebold) [inline src/components/metadata/metadata.js:206]: Is there a difference between `runCommand` and `metadata?.runCommand`?

I ask because I wonder if you can do `Boolean(runCommand)` for your new `isCommand` prop, the way it's done for the `visible` prop above?

Comment (rashidakanchwala) [reply src/components/metadata/metadata.js:206]: yeah, if you go top of the file. u will see we if there's no metadata?.runCommand we specify the runCommand to be that test.

File: RELEASE.md [hunk 4]
```diff
@@ -8,6 +8,14 @@ Please follow the established format:
 
 # Next release
 
+## Major features and improvements
+
+- Enable the display of json objects with react-json-viewer in the Experiment Tracking Overview Section. (#1236)
```
Comment (tynandebold) [inline RELEASE.md:13]:
```suggestion
- Enable the display of json objects with `react-json-viewer` in the experiment tracking. (#1236)
```

File: package/kedro_viz/models/flowchart.py [hunk 5]
```diff
@@ -427,7 +427,7 @@ def __post_init__(self, task_node: TaskNode):
         # if a node doesn't have a user-supplied `_name` attribute,
         # a human-readable run command `kedro run --to-nodes/nodes` is not available
         if kedro_node._name is not None:
-            self.run_command = f'kedro run --to-nodes="{kedro_node._name}"'
```
Comment (jmholzer) [inline package/kedro_viz/models/flowchart.py:430]: Why is it important to remove these quotes?

File: demo-project/conf/base/catalog_09_tracking.yml [hunk 6]
```diff
@@ -1,9 +1,9 @@
-train_evaluation.linear_regression:
+train_evaluation.linear_regression.r2_score:
```
Comment (jmholzer) [inline demo-project/conf/base/catalog_09_tracking.yml:1]: Why does this need to be changed? Was this a problem before?

File: demo-project/src/demo_project/pipelines/data_ingestion/nodes.py [hunk 7]
```diff
@@ -96,7 +96,7 @@ def aggregate_company_data(typed_companies: pd.DataFrame) -> pd.DataFrame:
 
 
 def combine_shuttle_level_information(
-    shuttles: pd.DataFrame, companies: pd.DataFrame, reviews: pd.DataFrame
+    shuttles: pd.DataFrame, shuttles2:pd.DataFrame, companies: pd.DataFrame, reviews: pd.DataFrame
```
Comment (jmholzer) [inline demo-project/src/demo_project/pipelines/data_ingestion/nodes.py:99]: Is there a more descriptive name we can use than `shuttles2`? Having `shuttles` and `shuttles2` is inconsistent and doesn't help me distinguish why there are two.

File: demo-project/src/demo_project/pipelines/data_ingestion/pipeline.py [hunk 8]
```diff
@@ -47,7 +50,8 @@ def create_pipeline(**kwargs) -> Pipeline:
             node(
                 func=combine_shuttle_level_information,
                 inputs={
-                    "shuttles": "int_typed_shuttles",
+                    "shuttles": "int_typed_shuttles@pandas2",
+                    "shuttles2": "int_typed_shuttles@pandas3",
```
Comment (jmholzer) [inline demo-project/src/demo_project/pipelines/data_ingestion/pipeline.py:54]: Same comment as above.

File: package/kedro_viz/launchers/cli.py [hunk 9]
```diff
@@ -20,7 +20,7 @@ def commands():  # pylint: disable=missing-function-docstring
     pass
 
 
-@commands.command(context_settings=dict(help_option_names=["-h", "--help"]))
+@commands.command(context_settings={"help_option_names": ["-h", "--help"]})
```
Comment (jmholzer) [inline package/kedro_viz/launchers/cli.py:23]: Was the linter complaining about this?

**Comment by rashidakanchwala** [comment]
> In general, we should open one PR per issue. It is very difficult for me to see which changes are relevant to which issue in this PR, though even if it were more obvious, constantly tracing back makes reviewing more difficult.
> 
> Unfortunately, I don't have the time to manually test these changes at the moment.

agree, it's just these bugs came up while trying to solve for run command stuff. so just thought of handling them here in one PR.

File: demo-project/src/demo_project/pipelines/data_ingestion/nodes.py (hunk 7 above)
Comment (rashidakanchwala) [reply demo-project/src/demo_project/pipelines/data_ingestion/nodes.py:99]: hey Jannic, i mentioned in my QA notes, this is just for front-end test purpose. and this wont be merged. Infact i will remove it now since Tynan approved the fe bit

File: demo-project/conf/base/catalog_09_tracking.yml (hunk 6 above)
Comment (rashidakanchwala) [reply demo-project/conf/base/catalog_09_tracking.yml:1]: it was wrong before and I am not sure since how long. on the pipeline the dataset is called r2_score but because it's not correctly defined in the catalog. these were getting stored as MemoryDataSets.

File: package/kedro_viz/models/flowchart.py (hunk 5 above)
Comment (rashidakanchwala) [reply package/kedro_viz/models/flowchart.py:430]: Ahdra mentioned as per the new cli commands we don't require quotes.

File: package/kedro_viz/launchers/cli.py (hunk 9 above)
Comment (rashidakanchwala) [reply package/kedro_viz/launchers/cli.py:23]: yes

**Review by jmholzer — APPROVED** [review]
✅
